public/static/partials/user_host_details.html

```diff
@@ -51,9 +51,10 @@ <h2 style="display:inline-block; padding-right:15px">
       </section>
       <section layout="row" flex style="white-space: nowrap">
           <div flex="85">
-            <strong>Expires On:</strong>
-            <md-datepicker ng-model="curHostData.date_for_expiration" md-min-date="curHostData.time_for_expiration" md-placeholder="Date"></md-datepicker>
-            <md-time-picker class="inline" ng-model="curHostData.time_for_expiration" message="message"></md-time-picker>
+            <strong>Expiration:</strong>
+            <md-datepicker ng-show="!curHostData.no_expiration" md-hide-icons="calendar" ng-model="curHostData.date_for_expiration" md-min-date="curHostData.time_for_expiration" md-placeholder="Date"></md-datepicker>
```

Comment:
Instead of hiding the date controls when you check no expiration, could you disable them? I feel that it gives a better user experience, since you don't have things disappearing on you

Reply:
It is jarring to have it disappear on you I agree, but disabling it just makes the expiration date un-editable (like it still shows 10/17/19 or etc), and we don't want any date to show. I can move the Never to the beginning of the line so only the end part drops off maybe.

Reply:
I do have a question about how to keep the formatting the same when they disappear.

Reply:
You could clear the data as well. I think the never expire checkbox makes more sense on the right, but it'd be nice to disable rather than hide